File: 3 КУРС/2 семестр/Учебная практика/JavaProects/app/src/main/java/com/example/voiceassistent/Message.java
package com.example.voiceassistent;

import android.graphics.Bitmap;
import java.util.ArrayList;
import java.util.Date;

public class Message {

    public String text;
    public Date date;
    public Boolean isSend;
    public ArrayList <String> str;
    public Message message;
    public Bitmap image;

    public Message(String text, boolean isSend) {

        this.text = text;
        this.isSend = isSend;
        this.date = new Date();
    }

    public Message(ArrayList<String> str, boolean isSend) {
        this.str = str;
        this.isSend = isSend;
        this.date = new Date();
    }


    public Message(MessageEntity entity) {
        this.text = entity.text;
        if (entity.date == null)
            this.date = null;
        else
            this.date = new Date(entity.date);
        this.isSend = entity.isSend != 0;
    }


    public Message(Bitmap image, boolean isSend) {
        this.image = image;
        this.isSend = isSend;
        this.date = new Date();
    }


}

File: 3 КУРС/2 семестр/Учебная практика/asd/JavaProects/app/src/main/java/com/example/voiceassistent/Message.kt
package com.example.voiceassistent

import android.graphics.Bitmap
import java.util.*

class Message {
    @JvmField
    var text: String? = null
    @JvmField
    var date: Date? = null
    @JvmField
    var isSend: Boolean
    var str: ArrayList<String>? = null
    var message: Message? = null
    var image: Bitmap? = null

    constructor(text: String?, isSend: Boolean) {
        this.text = text
        this.isSend = isSend
        date = Date()
    }

    constructor(str: ArrayList<String>?, isSend: Boolean) {
        this.str = str
        this.isSend = isSend
        date = Date()
    }

    constructor(entity: MessageEntity) {
        text = entity.text
        if (entity.date == null) date = null else date = Date(entity.date)
        isSend = entity.isSend != 0
    }

    constructor(image: Bitmap?, isSend: Boolean) {
        this.image = image
        this.isSend = isSend
        date = Date()
    }
}